src/Abuse/Adapters/TimeLimit.php

```diff
@@ -232,11 +231,27 @@ protected function hit(string $key, string $datetime): void
                     'count' => 1,
                     '$collection' => TimeLimit::COLLECTION,
                 ];
-                $this->db->createDocument(TimeLimit::COLLECTION, new Document($data));
+
+                try {
+                    $this->db->createDocument(TimeLimit::COLLECTION, new Document($data));
+                } catch (Duplicate $e) {
+                    // Duplicate in case of race condition
+                    /** @var Document $data */
+                    $data = $this->db->findOne(TimeLimit::COLLECTION, [
+                        Query::equal('key', [$key]),
+                        Query::equal('time', [$datetime]),
+                    ]);
+
+                    if ($data != false) {
+                        $this->count = intval($data->getAttribute('count'));
```

Comment:
Can't grabbing the count here like this possibly lead to an incorrect value? Suppose multiple concurrent calls are made, and all hit this at the same time. They'll all return the same data, right? But the `increaseDocumentAttribute()` could increment this to a value much higher than `$this->count++`;

Reply:
Yes I think this will solve the race condition for 2 concurrent calls, on 3 concurrent calls I think you are right.
you can get a false result but at least not an error Thrown

Reply:
See https://github.com/utopia-php/abuse/pull/37#issuecomment-1459930027